### DISCUSSION TITLE
How to compare 2 point clouds

### DISCUSSION THREAD

**Discussion by jon-rayo** [isl-org/Open3D#5810] [General]
Hello, I am new to open3d. I have two point clouds and would like to compare the 2. Like find the distance between the 2 point clouds. I just need general guidance on what I should use to compare 2 point clouds. It is 2 point clouds of the same object but with one difference. I would like to find the distance of the points in order to find the difference. Any help would be appreciated.

**Comment by qiutiantxwd** [comment]
You can try the compute_point_cloud_distance() which can be used to compute the Chamfer distance between 2 point clouds.
See documentation: (http://www.open3d.org/docs/latest/tutorial/geometry/pointcloud.html?highlight=chamfer)

**Reply by csyhping** [reply]
Hi @qiutiantxwd , may I ask if this distance from `compute_point_cloud_distance() ` is squared distance?

**Reply by qiutiantxwd** [reply]
Hello, by definition of chamfer distance, it should have the dimension of distance, not squared distance. 
But to make sure of this, you probably have to look into the source of this function. 
If you look at [this line](https://github.com/isl-org/Open3D/blob/48ccf2a4ea0471919fae2cb42d8d0889bb9562e2/cpp/open3d/geometry/PointCloud.cpp#L141), it looks like kdtree.SearchKNN returns squared distance, and this line of code does a sqrt on it to convert back to distance. 
I haven't used this in a while though, you might need to dig more into the source.
